pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -175,7 +175,9 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	}
 	// all secret events before this signal must be processed before we're marked "ready"
 	c.queue.Add(initialSyncSignal)
-	wait.Until(c.runWorker, 5*time.Second, stopCh)
+
+	go wait.Until(c.runWorker, 5*time.Second, stopCh)
```

Comment:
this follows how k8s controllers do things. queue is in routine, block on stopch, when it exits the `defers` above clean things up.